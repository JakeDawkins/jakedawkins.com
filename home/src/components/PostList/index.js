import React, { PropTypes } from "react";
import PostListItem from "../PostListItem";

let index = 0;

const renderPostListItems = (posts) => {
  return posts.map(post => {
    return <PostListItem key={index++} {...post} />
  });
}

const PostList = (props) =>
  <section className="content">
    <main id="scroll" className="home" role="main" itemProp="mainContentOfPage" itemScope="itemscope" itemType="http://schema.org/Blog">
      {renderPostListItems(props.posts)}
    </main>
  </section>

PostList.propTypes = {
  posts: PropTypes.array,
}

export default PostList;
