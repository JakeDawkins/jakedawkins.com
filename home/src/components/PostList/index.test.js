import PostList from './';
import React from 'react';
import renderer from 'react-test-renderer';

const posts = [{
    postId: 0,
    publishedDateTime: "1475459431",
    publishedDay: "02",
    publishedMonth: "October",
    publishedYear: "2016",
    linkUrl: "/test", //relative link
    title: "tester",
    description: "very test much wow",
    readMoreText: "View on GitHub...", //title ex: "read more..."
    tags: ["web","react"],
    content: "madness",
  },{
    postId: 1,
    publishedDateTime: "1475459431",
    publishedDay: "02",
    publishedMonth: "October",
    publishedYear: "2016",
    linkUrl: "/test2", //relative link
    title: "tester2",
    description: "very test much wow 2",
    readMoreText: "View on GitHub2...", //title ex: "read more..."
    tags: ["web"],
    content: "madness2",
  }];

describe("PostList", () => {

  it("should exist", () => {
    const postList = renderer.create(
      <PostList posts={[]} />
    );

    expect(postList).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    // const postList = renderer.create(
    //   <PostList posts={[]} />
    // );

    const postListWithProps = renderer.create(
      <PostList
        posts={posts}
      />
    );

    expect(postListWithProps).toMatchSnapshot();
    // expect(postListWithProps).not.toBe(postList);
  });

});
