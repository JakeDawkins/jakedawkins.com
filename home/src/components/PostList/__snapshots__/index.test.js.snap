exports[`PostList should exist 1`] = `
<section
  className="content">
  <main
    className="home"
    id="scroll"
    itemProp="mainContentOfPage"
    itemScope="itemscope"
    itemType="http://schema.org/Blog"
    role="main" />
</section>
`;

exports[`PostList should render props correctly 1`] = `
<section
  className="content">
  <main
    className="home"
    id="scroll"
    itemProp="mainContentOfPage"
    itemScope="itemscope"
    itemType="http://schema.org/Blog"
    role="main">
    <article
      className="post-item"
      itemProp="blogPost"
      itemScope="itemscope"
      itemType="http://schema.org/BlogPosting"
      role="article">
      <a
        className="datetime"
        onClick={[Function bound handleClick]}
        style={Object {}}>
        <meta
          content="1475459431"
          itemProp="datePublished" />
        <time
          dateTime="1475459431">
          <span
            className="day">
             
            02
             
          </span>
          <span
            className="month-year">
             
            October
             
            2016
             
          </span>
        </time>
      </a>
      <div
        className="content">
        <a
          onClick={[Function bound handleClick]}
          style={Object {}}>
          <h2
            className="post-title"
            itemProp="name">
            tester
          </h2>
        </a>
        <p
          className="description">
          very test much wow 
          <a
            className="link"
            onClick={[Function bound handleClick]}
            style={Object {}}>
            View on GitHub...
          </a>
        </p>
        <div
          className="tags" />
      </div>
    </article>
    <article
      className="post-item"
      itemProp="blogPost"
      itemScope="itemscope"
      itemType="http://schema.org/BlogPosting"
      role="article">
      <a
        className="datetime"
        onClick={[Function bound handleClick]}
        style={Object {}}>
        <meta
          content="1475459431"
          itemProp="datePublished" />
        <time
          dateTime="1475459431">
          <span
            className="day">
             
            02
             
          </span>
          <span
            className="month-year">
             
            October
             
            2016
             
          </span>
        </time>
      </a>
      <div
        className="content">
        <a
          onClick={[Function bound handleClick]}
          style={Object {}}>
          <h2
            className="post-title"
            itemProp="name">
            tester2
          </h2>
        </a>
        <p
          className="description">
          very test much wow 2 
          <a
            className="link"
            onClick={[Function bound handleClick]}
            style={Object {}}>
            View on GitHub2...
          </a>
        </p>
        <div
          className="tags" />
      </div>
    </article>
  </main>
</section>
`;
